File: app/src/main/java/com/eugene_poroshin/money_manager/repo/database/CategoryDao.java
package com.eugene_poroshin.money_manager.repo.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface CategoryDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(CategoryEntity category);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void update(CategoryEntity category);

    @Delete
    void delete(CategoryEntity category);

    @Query("SELECT * FROM categories")
    LiveData<List<CategoryEntity>> getAllCategories();

    @Query("SELECT category_name FROM categories")
    LiveData<List<String>> getCategoryNames();
}

File: app/src/main/java/com/eugene_poroshin/money_manager/repo/database/CategoryDao.kt
package com.eugene_poroshin.money_manager.repo.database

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface CategoryDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(category: CategoryEntity)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    suspend fun update(category: CategoryEntity)

    @Delete
    suspend fun delete(category: CategoryEntity)

    @Query("SELECT * FROM categories")
    fun allCategories(): LiveData<List<CategoryEntity>>
}